uniswap-v2/src/modules/13_store_daily_and_hourly_fields.rs

```diff
@@ -0,0 +1,99 @@
+use substreams::pb::substreams::Clock;
+use substreams::store::{DeltaBigDecimal, Deltas, StoreAdd};
+use substreams::store::{StoreAddBigDecimal, StoreNew};
+
+use crate::store_key::StoreKey;
+use crate::utils;
+
+#[substreams::handlers::store]
+pub fn store_daily_and_hourly_fields(
+    clock: Clock,
+    volume_deltas: Deltas<DeltaBigDecimal>,
+    output_store: StoreAddBigDecimal,
+) {
+    let timestamp = clock.timestamp.unwrap().seconds;
+
+    let day_id = utils::get_day_id(timestamp);
+    let hour_id = utils::get_hour_id(timestamp);
+
+    for delta in volume_deltas.deltas {
+        let ordinal = delta.ordinal;
+        let volume = utils::delta_value(&delta);
+
+        if let Some(pool_address) = StoreKey::Volume.get_pool(&delta.key) {
+            let (supply_side_revenue, protocol_side_revenue) =
+                utils::calculate_revenue(volume.clone());
+
+            output_store.add(
+                ordinal,
+                StoreKey::HourlyVolumeUSD
+                    .get_unique_hourly_pool_key(hour_id.clone(), &pool_address),
```

Comment:
Have already added comment about `StoreKey::HourlyVolumeUSD.get_unique_hourly_pool_key` so make sure to remove all these clones once fn is modded and change these to refs instead :)